mkdocs/utils/__init__.py

```diff
@@ -431,7 +435,10 @@ def nest_paths(paths):
 
         branch = nested
         for part in parts:
-            part = filename_to_title(part)
+            dir = False
+            if os.path.basename(part) != path:
+                dir = True
```

Comment:
Ah, okay. I wasn't sure if the function was called from anywhere else in the code. Then we definitely need two separate code paths, one for filenames and one for dirnames.